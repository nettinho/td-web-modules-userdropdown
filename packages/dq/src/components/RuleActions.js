import _ from "lodash/fp";
import React from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { compose, bindActionCreators } from "redux";
import PropTypes from "prop-types";
import { Button } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { ConfirmModal } from "@truedat/core/components";
import { linkTo } from "../routes";
import { deleteRule } from "../routines";

const createRuleImplementationUrl = linkTo.RULE_IMPLEMENTATION_NEW;

const getCreateRuleUrl = ({ id }) => createRuleImplementationUrl({ id });

const RuleActions = ({ deleteRule, rule: { id, name }, match }) => (
  <div style={{ float: "right" }}>
    <Button
      secondary
      as={Link}
      to={getCreateRuleUrl(match.params, id)}
      content={<FormattedMessage id="quality.rule.actions.create" />}
    />
    <Button
      secondary
      as={Link}
      to={linkTo.RULE_EDIT({ id })}
      content={<FormattedMessage id="quality.actions.edit" />}
    />
    <ConfirmModal
      icon="trash"
      trigger={
        <Button
          secondary
          content={<FormattedMessage id="quality.actions.remove" />}
        />
      }
      header={
        <FormattedMessage id="rules.actions.delete.confirmation.header" />
      }
      size="small"
      content={
        <FormattedMessage
          id="rules.actions.delete.confirmation.content"
          values={{ name: <i>{name}</i> }}
        />
      }
      handleSubmit={() => deleteRule({ id })}
    />
  </div>
);

RuleActions.propTypes = {
  deleteRule: PropTypes.func,
  rule: PropTypes.object.isRequired,
  match: PropTypes.object
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ deleteRule }, dispatch)
});

const mapStateToProps = ({ rule }) => ({
  rule
});

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(RuleActions);
