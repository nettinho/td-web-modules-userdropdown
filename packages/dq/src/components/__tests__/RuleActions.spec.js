import React from "react";
import { shallowWithIntl } from "intl-stub";
import RuleActions from "../RuleActions";

describe("<RuleActions />", () => {
  const rule = { id: 2, name: "foo", business_concept_id: 1 };
  const match = { params: { id: 1, qc_id: 2 } };
  const concepts = [
    {
      id: 1,
      name: "s1",
      description: "dd",
      version: "vs",
      status: "st",
      business_concept_id: 1
    }
  ];
  const props = { rule, concepts, match };
  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<RuleActions {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
