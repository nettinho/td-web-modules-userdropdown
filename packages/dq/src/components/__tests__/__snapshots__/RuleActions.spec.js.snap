// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RuleActions /> matches the latest snapshot 1`] = `<Route />`;
